libs/utils/src/extract_cdf_ui_strings.test.ts

```diff
@@ -0,0 +1,180 @@
+        },
+      }),
+    });
+  },
+
+  [ElectionStringKey.CANDIDATE_NAME]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.CONTEST_DESCRIPTION]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.CONTEST_OPTION_LABEL]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.CONTEST_TITLE]() {
+    const uiStrings = extractCdfUiStrings({
+      ...testCdfBallotDefinition,
+      Election: [
+        {
+          ...assertDefined(testCdfBallotDefinition.Election[0]),
+          Contest: [
+            {
+              ...assertDefined(testCdfBallotDefinition.Election[0]?.Contest[0]),
+              '@id': 'contest1',
+              BallotTitle: buildInternationalizedText({
+                [LanguageCode.ENGLISH]: 'President',
+                [LanguageCode.CHINESE]: '總統',
+                unsupported_lang: '🗳✅',
+              }),
+            },
+            {
+              ...assertDefined(testCdfBallotDefinition.Election[0]?.Contest[1]),
+              '@id': 'contest2',
+              BallotTitle: buildInternationalizedText({
+                [LanguageCode.ENGLISH]: 'Mayor',
+                [LanguageCode.CHINESE]: '市長',
+                unsupported_lang: '🗳✅',
+              }),
+            },
+          ],
+        },
+      ],
+    });
+
+    expect(uiStrings).toEqual({
+      [LanguageCode.ENGLISH]: expect.objectContaining({
+        [ElectionStringKey.CONTEST_TITLE]: {
+          contest1: 'President',
+          contest2: 'Mayor',
+        },
+      }),
+      [LanguageCode.CHINESE]: expect.objectContaining({
+        [ElectionStringKey.CONTEST_TITLE]: {
+          contest1: '總統',
+          contest2: '市長',
+        },
+      }),
+    });
+  },
+
+  [ElectionStringKey.COUNTY_NAME]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.DISTRICT_NAME]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.ELECTION_TITLE]() {
+    const uiStrings = extractCdfUiStrings({
+      ...testCdfBallotDefinition,
+      Election: [
+        {
+          ...assertDefined(testCdfBallotDefinition.Election[0]),
+          Name: buildInternationalizedText({
+            [LanguageCode.ENGLISH]: 'General Election',
+            [LanguageCode.SPANISH]: 'Elección General',
+            unsupported_lang: '🗳✅',
+          }),
+        },
+      ],
+    });
+
+    expect(uiStrings).toEqual({
+      [LanguageCode.ENGLISH]: expect.objectContaining({
+        [ElectionStringKey.ELECTION_TITLE]: 'General Election',
+      }),
+      [LanguageCode.SPANISH]: expect.objectContaining({
+        [ElectionStringKey.ELECTION_TITLE]: 'Elección General',
+      }),
+    });
+  },
+
+  [ElectionStringKey.PARTY_NAME]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.PRECINCT_NAME]() {
+    // TODO(kofi): Implement
+  },
+
+  [ElectionStringKey.STATE_NAME]() {
+    // TODO(kofi): Implement
+  },
+};
+
+for (const [electionStringKey, testFn] of Object.entries(tests)) {
+  test(electionStringKey, testFn);
+}
```

Comment:
I've been a fan of [`test.each` for repetitive tests](https://jestjs.io/docs/api#testeachtablename-fn-timeout) like these. This explicit loop is also totally fine! But just thought I'd note

Reply:
Oh, love that! TIL - thanks!